apps/web/src/pages/ApplicantDashboardPage/components/MetadataDate.tsx

```diff
@@ -0,0 +1,152 @@
+import { useIntl } from "react-intl";
+
+import { formatDate, parseDateTimeUtc } from "@gc-digital-talent/date-helpers";
+import { commonMessages } from "@gc-digital-talent/i18n";
+
+import { ApplicationStatus } from "~/utils/poolCandidate";
+
+export const ApplicationDate = ({
+  closingDate,
+  submittedAt,
+  finalDecisionAt,
+  status,
+}: {
+  closingDate?: string | null;
+  submittedAt?: string | null;
+  finalDecisionAt?: string | null;
+  status: ApplicationStatus;
+}) => {
+  const intl = useIntl();
+  const nullMessage = intl.formatMessage(commonMessages.notFound);
+
+  const isDraftStatus = status === ApplicationStatus.DRAFT;
+  const isExpiredStatus = status === ApplicationStatus.EXPIRED;
+  if (isDraftStatus || isExpiredStatus) {
+    return (
+      <span>
+        {intl.formatMessage({
+          defaultMessage: "Deadline",
+          id: "nIAA4Q",
+          description: "Label for deadline metadata",
+        })}{" "}
+        {intl.formatMessage(commonMessages.dividingColon)}{" "}
+        {closingDate
+          ? formatDate({
+              date: parseDateTimeUtc(closingDate),
+              formatString: "PPP",
+              intl,
+              timeZone: "Canada/Pacific",
+            })
+          : nullMessage}
+      </span>
+    );
+  }
+
+  const isReceivedStatus = status === ApplicationStatus.RECEIVED;
+  const isUnderReviewStatus = status === ApplicationStatus.UNDER_REVIEW;
+  const isUnderAssessmentStatus = status === ApplicationStatus.UNDER_ASSESSMENT;
+  if (isReceivedStatus || isUnderReviewStatus || isUnderAssessmentStatus) {
+    return (
+      <span>
+        {intl.formatMessage({
+          defaultMessage: "Submitted",
+          id: "B1sXVl",
+          description: "Label for submitted metadata",
+        })}{" "}
+        {intl.formatMessage(commonMessages.dividingColon)}{" "}
+        {submittedAt
+          ? formatDate({
+              date: parseDateTimeUtc(submittedAt),
+              formatString: "PPP",
+              intl,
+              timeZone: "Canada/Pacific",
+            })
+          : nullMessage}
+      </span>
+    );
+  }
+
+  const isSuccessfulStatus = status === ApplicationStatus.SUCCESSFUL;
+  const isUnsuccessfulStatus = status === ApplicationStatus.UNSUCCESSFUL;
+  if (isSuccessfulStatus || isUnsuccessfulStatus) {
+    return (
+      <span>
+        {intl.formatMessage({
+          defaultMessage: "Assessed",
+          id: "GYcxDu",
+          description: "Label for assessed metadata",
+        })}{" "}
+        {intl.formatMessage(commonMessages.dividingColon)}{" "}
+        {finalDecisionAt
+          ? formatDate({
+              date: parseDateTimeUtc(finalDecisionAt),
+              formatString: "PPP",
+              intl,
+              timeZone: "Canada/Pacific",
+            })
+          : nullMessage}
+      </span>
+    );
+  }
+
+  return null;
```

Comment:
I feel like we could move this to the top of the file so we have an early return and do not need to run through all these checks just to return null.

This applies all of the other components that also return null.

Reply:
Changed in [74f9438](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12366/commits/74f94387d46c080ee8c94ee18897bdb729dcdacc)